File: src/main/js/funk/ioc/inject.js
funk.ioc = funk.ioc || {};
funk.ioc.inject = function(type){
    if(type == null) {
        throw new funk.error.ArgumentError("Given type must not be null.")
    }

    var currentScope = funk.ioc.Injector.currentScope();
    if(!funk.isValid(currentScope)){
        return funk.ioc.Injector.scopeOf(type).getInstance(type);
    }
    return currentScope.getInstance(type);
};

// Alias
var inject = funk.ioc.inject;

File: src/main/dart/funk/ioc/inject.dart
inject(Type type) {
  if(type == null) {
    throw new ArgumentError("Given type must not be null.");
  }
  
  final Injector injector = new Injector();
  final IModule currentScope = when(injector.currentScope, {
    "none": () => null,
    "some": (value) => value
  });

  if(null == currentScope) {
    return injector.scopeOf(type).getInstance(type);
  }

  return currentScope.getInstance(type);
}
